### DISCUSSION TITLE
Runtime parser configuration?

### DISCUSSION THREAD

**Discussion by diskdance** [tree-sitter/tree-sitter#3510] [Q&A] [answered]
Hello,

I'm working on parsing an awkward Markdown-like markup language which has a configurable list of "keywords". To fit into different configurations, it is needed to make the keywords list configurable at runtime. 

It looks like:

```
This is some text __KEYWORD1__ __RUNTIME_CONFIGURED_KEYWORD2__ 
<!-- Otherwise, __RUNTIME_CONFIGURED_KEYWORD2__ should be recognized as plain text -->
```

Is it possible using tree-sitter? Thank you.

**Comment by amaanq** [comment, marked as answer]
There's no such mechanism to make this configurable at runtime. You could do what [Markdown does](https://github.com/tree-sitter-grammars/tree-sitter-markdown/blob/7fe453beacecf02c86f7736439f238f5bb8b5c9b/common/grammar.js#L1) and have it be configurable at *generate* time though.
